apps/web/src/components/PublishMenu.tsx

```diff
@@ -37,6 +54,8 @@ export const PublishMenu: React.FC<PublishmenuProps> = ({ page }) => {
 
   const publishMutation = useMutation(
     async () => {
+      page.data = await onPublishContent(page.data, encryptionKey);
```

Comment:
I suspect this mutation of props can introduce annoying bugs 🐞 maybe we can make it immutable?